ocs_ci/ocs/ui/performance_lib_ui.py

```diff
@@ -0,0 +1,243 @@
+import logging
+
+from selenium.webdriver.common.by import By
+from ocs_ci.ocs.exceptions import TimeoutException
+from ocs_ci.ocs.ui.base_ui import SeleniumDriver
+from selenium.webdriver.support.ui import WebDriverWait
+import time
+from selenium.webdriver.support import expected_conditions as ec
+from selenium.webdriver.common.action_chains import ActionChains
+
+
+logger = logging.getLogger(__name__)
+
+
+def grafana_resource_consumption_ui(test_duration, url, username, password):
+    """
+    This function logs in to grafana and captures resource consumption during performance test executions
+
+    Args:
+    test_duration (int): Test duration, for which the resource consumption needs to be captured
+    url (str) : Grafana url
+    username (str) : grafana username
+    password (str): grafana password
+
+    """
+    driver = SeleniumDriver()
+
+    if test_duration == 0:
+        logger.error("The test duration value cannont be zero")
+    elif test_duration > 0 and test_duration < 301:
+        last_duration = "Last 5 minutes"
+    elif test_duration > 300 and test_duration < 901:
+        last_duration = "Last 15 minutes"
+    elif test_duration > 900 and test_duration < 1801:
+        last_duration = "Last 30 minutes"
+    elif test_duration > 1800 and test_duration < 3601:
+        last_duration = "Last 1 hour"
+    elif test_duration > 3600 and test_duration < 10801:
+        last_duration = "Last 3 hour"
+    elif test_duration > 10800 and test_duration < 21601:
+        last_duration = "Last 6 hour"
+    elif test_duration > 21600 and test_duration < 43201:
+        last_duration = "Last 12 hour"
+    elif test_duration > 43200 and test_duration < 86401:
+        last_duration = "Last 24 hour"
+    else:
+        logger.error(f"The test duration value {test_duration} is invalid")
+    try:
```

Comment:
Under this try block all the click and capture operations are performed, I see we have finally block, good to have a exception as well WDYT?